x-pack/legacy/plugins/ml/public/data_visualizer/directive.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+// @ts-ignore
+import { uiModules } from 'ui/modules';
+const module = uiModules.get('apps/ml', ['react']);
+
+import { I18nContext } from 'ui/i18n';
+import { IndexPattern, IndexPatterns } from 'ui/index_patterns';
+import { IPrivate } from 'ui/private';
+import { InjectorService } from '../../common/types/angular';
+
+import { KibanaConfigTypeFix, KibanaContext } from '../contexts/kibana/kibana_context';
+// @ts-ignore
+import { SearchItemsProvider } from '../jobs/new_job/utils/new_job_utils';
+// Simple drop-in type until new_job_utils offers types.
+type CreateSearchItems = () => {
+  indexPattern: IndexPattern;
+  savedSearch: any;
+  combinedQuery: any;
+};
+
+import { Page } from './page';
+
+module.directive('mlDataVisualizer', ($injector: InjectorService) => {
+  return {
+    scope: {},
+    restrict: 'E',
+    link: (scope: ng.IScope, element: ng.IAugmentedJQuery) => {
+      const indexPatterns = $injector.get<IndexPatterns>('indexPatterns');
+      const kbnBaseUrl = $injector.get<string>('kbnBaseUrl');
+      const kibanaConfig = $injector.get<KibanaConfigTypeFix>('config');
+      const Private: IPrivate = $injector.get('Private');
```

Comment:
could be `const Private = $injector.get<IPrivate>('Private');`

Reply:
Done in https://github.com/elastic/kibana/pull/42685/commits/992c67e94e727b7b5b5f91cbc73eb954c6e13a0f